rootfs/api/models/release.py

```diff
@@ -136,36 +134,38 @@ def previous(self):
             prev_release = None
         return prev_release
 
-    def rollback(self, user, version):
-        if version < 1:
-            raise EnvironmentError('version cannot be below 0')
-
-        summary = "{} rolled back to v{}".format(user, version)
-        prev = self.app.release_set.get(version=version)
-        new_release = self.new(
-            user,
-            build=prev.build,
-            config=prev.config,
-            summary=summary,
-            source_version='v{}'.format(version)
-        )
-
+    def rollback(self, user, version=None):
         try:
+            # if no version is provided then grab version from object
+            version = (self.version - 1) if version is None else int(version)
+
+            if version < 1:
+                raise DeisException('version cannot be below 0')
+
+            prev = self.app.release_set.get(version=version)
+            new_release = self.new(
+                user,
+                build=prev.build,
+                config=prev.config,
+                summary="{} rolled back to v{}".format(user, version),
+                source_version='v{}'.format(version)
+            )
+
             if self.build is not None:
                 self.app.deploy(new_release)
             return new_release
-        except Exception:
+        except Exception as e:
             if 'new_release' in locals():
                 new_release.delete()
-            raise
+            raise DeisException(str(e))
```

Comment:
It doesn't mask it in any meaningful way, also we never tried catching any of the exceptions you mentions and rarely tried to catch the top level Exception.

We need to not let random exceptions leak out to the user with a massive trace back. We need to handle them